### DISCUSSION TITLE
[Bug]:  404 Not Found: new chat

### DISCUSSION THREAD

**Discussion by kuangxiaoye** [danny-avila/LibreChat#1715] [Troubleshooting]
### What happened?

"http://localhost:3080/c/new" is constantly in a 404 status.

### Steps to Reproduce

I am very sorry, even after multiple deployments, the it still cannot be successfully run...

1. npm i
2. npm run frontend
3. npm run backend

 I have enabled .env of debug and reported the issue with a screenshot.

### Relevant log output

```shell
(base) wangye@wangyedeMac-mini LibreChat % npm run backend

> LibreChat@0.6.6 backend
> cross-env NODE_ENV=production node api/server/index.js

2024-02-03 16:32:52 info: [Optional] Redis not initialized. Note: Redis support is experimental.
2024-02-03 16:32:53 warn: Meilisearch error, search will be disabled
2024-02-03 16:32:54 info: Connected to MongoDB
2024-02-03 16:32:54 info: Server listening at http://localhost:3080
2024-02-03 16:32:56 debug: [validateJson] files
2024-02-03 16:32:59 debug: [validateJson] files
```

### Screenshots

<img width="1916" alt="image" src="https://github.com/danny-avila/LibreChat/assets/43713316/11bcab45-ad44-464d-9295-265baf65ee56">

<img width="749" alt="image" src="https://github.com/danny-avila/LibreChat/assets/43713316/f2c19c2c-7f8c-4f10-8e09-bf576e373d82">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
You are highlighting 2 issues here.

> is constantly in a 404 status.

But your screenshot doesn't show this?

For the screen issue, try updating chrome. This happens on old browser that do not support dynamic viewport.

This happens as a result of this PR: https://github.com/danny-avila/LibreChat/pull/1608

I will have a fix to prevent it altogether, but you may be able to fix it by updating.

**Reply by danny-avila** [reply]
It is running completely fine for me after updating with the update script: 

![image](https://github.com/danny-avila/LibreChat/assets/110412045/20ba3cdd-1284-4be6-a63c-f94fda54b95b)

 If you're having issues running this command, you can try running what the script does manually: 

```bash
# Stop the container (if running)
docker-compose down
# Switch to the repo's main branch
git checkout main
# Pull the latest changes to the main branch from Github
git pull 
# Prune all LibreChat Docker images
docker rmi librechat:latest
# Remove all unused dangling Docker images.
# Be careful, as this will delete all dangling docker images on your
# computer, also those not created by LibreChat!
docker image prune -f
# Building a new LibreChat image without cache
docker-compose build --no-cache

# Start LibreChat
docker-compose up
```

**Comment by mbaroukh** [comment]
Hello.
I too encounter the exact same problem, running either from source or from the latest docker build.
NOTE: Even i the server returns a 404, the apps starts correctly because the index.html file is returned. But I have in the browser from times to times a message saying "oops, the page doesnt't xist anymore. Would you like to use webarchive ?"

I suppose, the reason is just because of this line : 

https://github.com/danny-avila/LibreChat/blob/0e3e45e77dd80a18a358007a2601a03be1084208/api/server/index.js#L96

It always returns a 404 with the index of the react front app.
I don't get why it is a 404. or me it should be a 200 and the 404 should be handle on the client by react-router.
With a 200 instead of 404 I don't have the problem anymore.

**Reply by danny-avila** [reply]
Thanks @mbaroukh this seems like a legacy thing that got overlooked early on in the project. You're right to let the client-side routing do its job. Making the change here: https://github.com/danny-avila/LibreChat/pull/3010/commits/876260b92b36326d28c06acef295375d12ded05b

**Comment by mbaroukh** [comment]
Thanks a lot for your reply and your job.
Librechat is amazing

**Reply by danny-avila** [reply]
thank you! glad you're enjoying it!
